python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -0,0 +1,341 @@
+ */
+predicate jumpStep(ExprNode pred, ExprNode succ) {
+  // As we have ESSA variables for global variables,
+  // we include ESSA flow steps involving global variables.
+  (
+    pred.asEssaNode() instanceof GlobalSsaVariable
+    or
+    succ.asEssaNode() instanceof GlobalSsaVariable
+  ) and
+  EssaFlow::essaFlowStep(pred, succ)
+}
+
+//--------
+// Field flow
+//--------
+
+/**
+ * Holds if data can flow from `node1` to `node2` via an assignment to
+ * content `c`.
+ */
+predicate storeStep(Node node1, Content c, Node node2) {
+  none()
+}
+
+/**
+ * Holds if data can flow from `node1` to `node2` via a read of content `c`.
+ */
+predicate readStep(Node node1, Content c, Node node2) {
+  none()
+}
+
+/**
+ * Holds if values stored inside content `c` are cleared at node `n`. For example,
+ * any value stored inside `f` is cleared at the pre-update node associated with `x`
+ * in `x.f = newValue`.
+ */
+cached
+predicate clearsContent(Node n, Content c) {
+  none()
+}
+
+//--------
+// Fancy context-sensitive guards
+//--------
+
+/**
+ * Holds if the node `n` is unreachable when the call context is `call`.
+ */
+predicate isUnreachableInCall(Node n, DataFlowCall call) {
+  none()
+}
+
+//--------
+// Fancy dispatch
+//--------
+
+/**
+ * Holds if the call context `ctx` reduces the set of viable run-time
+ * targets of call `call` in `c`.
+ */
+predicate reducedViableImplInCallContext(DataFlowCall call, DataFlowCallable c, DataFlowCall ctx) {
+  none()
+}
+
+/**
+ * Holds if flow returning from callable `c` to call `call` might return
+ * further and if this path restricts the set of call sites that can be
+ * returned to.
+ */
+predicate reducedViableImplInReturn(DataFlowCallable c, DataFlowCall call) {
+  none()
+}
+
+/**
+ * Gets a viable run-time target for the call `call` in the context `ctx`.
+ * This is restricted to those call nodes and results for which the return
+ * flow from the result to `call` restricts the possible context `ctx`.
+ */
+DataFlowCallable prunedViableImplInCallContextReverse(DataFlowCall call, DataFlowCall ctx) {
+  none()
+  // result = viableImplInCallContext(call, ctx) and
+  // reducedViableImplInReturn(result, call)
+}
+
+/**
+ * Gets a viable run-time target for the call `call` in the context
+ * `ctx`. This is restricted to those call nodes for which a context
+ * might make a difference.
+ */
+DataFlowCallable prunedViableImplInCallContext(DataFlowCall call, DataFlowCall ctx) {
+  none()
+  // result = viableImplInCallContext(call, ctx) and
+  // reducedViableImplInCallContext(call, _, ctx)
+}
+
+//--------
+// Misc
+//--------
+
+/**
+ * Holds if `n` does not require a `PostUpdateNode` as it either cannot be
+ * modified or its modification cannot be observed, for example if it is a
+ * freshly created object that is not saved in a variable.
+ *
+ * This predicate is only used for consistency checks.
+ */
+predicate isImmutableOrUnobservable(Node n) { none() }
+
+int accessPathLimit() { result = 3 }
```

Comment:
We've been using 5 as the default value (C# has lowered this to 3 due to otherwise bad field flow performance).

Reply:
Set to 5.